### ISSUE TITLE
A question about mw2 xmodel

### ISSUE THREAD

**Issue by dest1yo** [mauserzjeh/cod-asset-importer#7]
Sorry to disturb you. I would like to ask if you have researched the xmodel structure of mw2. Is there any relevant information that I can refer to?

**Comment by mauserzjeh** [comment]
I haven't research mw2 yet. Can the xmodels be extracted from the game files?

**Comment by dest1yo** [comment]
> I haven't research mw2 yet. Can the xmodels be extracted from the game files?

Yes i found that xmodels can be extracted from fast file by using Zonetool.
I did some research recently and learned a few things.

**Comment by dest1yo** [comment]
> I haven't research mw2 yet. Can the xmodels be extracted from the game files?

Thank you for your reply. Here is the Zonetool:
https://github.com/ZoneTool/zonetool
If you are interested, you can also take a look :)

**Comment by mauserzjeh** [comment]
Thanks! 

Currently I am rewriting the plugin in rust for better performance, as soon as I will finish supporting the already supported games, I will take a look at if I can support newer cods as well. I knew that after CoD4, the assets were mainly in modtools or fastfiles, but I haven't come across `zonetool`. This way I might be able to support maps as well.

No ETA yet though.

**Comment by dest1yo** [comment]
> Thanks!
> 
> Currently I am rewriting the plugin in rust for better performance, as soon as I will finish supporting the already supported games, I will take a look at if I can support newer cods as well. I knew that after CoD4, the assets were mainly in modtools or fastfiles, but I haven't come across `zonetool`. This way I might be able to support maps as well.
> 
> No ETA yet though.

You're welcome. Do you have any other contact information? Although I may not continue to research these for the time being.
By the way, do you mean that we can use rust to write plugins in blender?

**Comment by mauserzjeh** [comment]
There is a project called [Pyo3](https://pyo3.rs) which can build python module from rust code. I am writing most of the plugin in rust for better performance (asset loading, transforming data etc.). Then I build a python module from it (it's like a dll) which exports various functions that I can call from python. Feel free to check the `develop` branch, although its heavily work in progress.

**Comment by dest1yo** [comment]
> There is a project called [Pyo3](https://pyo3.rs) which can build python module from rust code. I am writing most of the plugin in rust for better performance (asset loading, transforming data etc.). Then I build a python module from it (it's like a dll) which exports various functions that I can call from python. Feel free to check the `develop` branch, although its heavily work in progress.

Cool! Good work!

**Comment by dest1yo** [comment]
> There is a project called [Pyo3](https://pyo3.rs) which can build python module from rust code. I am writing most of the plugin in rust for better performance (asset loading, transforming data etc.). Then I build a python module from it (it's like a dll) which exports various functions that I can call from python. Feel free to check the `develop` branch, although its heavily work in progress.

Another tool about ff. I haven't used it, it might be helpful to you
https://github.com/Laupetin/OpenAssetTools

**Comment by Aireania** [comment]
excuse me，how can i transplant a xmodel from waw 2 cod6
